trie/trie_test.go

```diff
@@ -74,175 +87,218 @@ func TestNewTrie(t *testing.T) {
 	}
 }
 
-type randTest struct {
+type trieTest struct {
 	key   []byte
 	value []byte
+	pk    []byte
+	op    int
 }
 
-func generateRandTest(size int) []randTest {
-	rt := make([]randTest, size)
+func generateRandTest(size int) []trieTest {
+	rt := make([]trieTest, size)
 	r := *rand.New(rand.NewSource(rand.Int63()))
 	for i := range rt {
-		rt[i] = randTest{}
+		rt[i] = trieTest{}
 		buf := make([]byte, r.Intn(379)+1)
 		r.Read(buf)
-		rt[i].key = buf
+		if !keyExists(rt[0:i], buf) {
+			rt[i].key = buf
 
-		buf = make([]byte, r.Intn(128))
-		r.Read(buf)
-		rt[i].value = buf
+			buf = make([]byte, r.Intn(128))
+			r.Read(buf)
+			rt[i].value = buf
+		}
 	}
 	return rt
 }
 
-func TestBranch(t *testing.T) {
-	trie := newEmpty()
+func keyExists(rt []trieTest, key []byte) bool {
+	for _, test := range rt {
+		if bytes.Equal(test.key, key) {
+			return true
+		}
+	}
 
-	key1 := []byte{0x01, 0x35}
-	value1 := []byte("spaghetti")
-	key2 := []byte{0x01, 0x35, 0x79}
-	value2 := []byte("gnocchi")
-	key3 := []byte{0x07}
-	value3 := []byte("ramen")
-	key4 := []byte{0xf2}
-	value4 := []byte("pho")
+	return false
+}
 
-	err := trie.Put(key1, value1)
-	if err != nil {
-		t.Errorf("Fail to put with key %x and value %x: %s", key1, value1, err.Error())
-	}
+func hexDecode(in string) []byte {
+	out, _ := hex.DecodeString(in)
+	return out
+}
 
-	err = trie.Put(key2, value2)
-	if err != nil {
-		t.Errorf("Fail to put with key %x and value %x: %s", key2, value2, err.Error())
+func writeToTestFile(tests []trieTest) error {
+	testString := ""
+	for _, test := range tests {
+		testString = fmt.Sprintf("%s%s\n%s\n", testString, test.key, test.value)
 	}
 
-	err = trie.Put(key3, value3)
+	fp, err := filepath.Abs("./test_data")
```

Comment:
Please rename this to something more indicative (easily confused with `gossamer_data`). Perhaps `failed_test_data`

Reply:
will do